src/main/java/core/mate/academy/service/TruckProducerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducerImpl implements MachineProducer<Truck> {

```

Comment:
It's good that you've parameterized `MachineProducer` with `Truck`, ensuring type safety.